### PR TITLE
fix(patterns/fileuploader): update documentation and add error cases

### LINKED ISSUE #610: fileUploader component
# 🐞 Pattern bug report

### Where did you experience this bug ?
	
- [x] On sketch file
- [x] On Documentation
- [x] On source code

### Is this a regression?

No

### Description

Regarding the implementation of forms, we use the uploader component but this component doesn't have any validation. To explain ourself, if user doesn't upload any file anf if it is require, we hava none indication of error like a little red sentense for example. Moreover, when you upload a file, if you have a bad upload, you have a little message for indicate this error, but it is in black an d not in red as the others design of message error in Mosaic.

uplo

### PR DESCRIPTION
**Pull request by tiloyi**
Resolves #610

## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

Update documentation and add error cases

GitHub issue number or Jira issue URL: [ISSUE-610](https://github.com/adeo/mozaic-design-system/issues/610)

### CODE REVIEW

File: src/docs/Components/Form/FileUploader/index.mdx [hunk 1]
```diff
@@ -26,18 +62,22 @@ After uploading a file, a tile is added to the bottom of the component. This til
 - Uploaded
```
Comment (nherwegh) [inline src/docs/Components/Form/FileUploader/index.mdx:62]: …is added **at** <s>to</s> the bottom of the component.

File: src/docs/Components/Form/FileUploader/code.mdx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+#### Accept type of files
+
+You can also restrict the type or the path of the file to be uploaded by using the `accept` attribute: `accept=".jpg, .jpeg, .png"` or `accept="image/*`.
+
+<Preview path="with-attribute-accept" />
+
+#### Display uploaded files and errors
+
+Html and css does not provide a native way to achieve this, but using some Javascript, it is possible to display errors and uploaded files to the user.
```
Comment (caqueste) [inline src/docs/Components/Form/FileUploader/code.mdx:63]: "Html and css does not provide" → I think in this case we have to write "Do" ?

File: src/docs/Components/Form/FileUploader/index.mdx [hunk 3]
```diff
@@ -26,18 +62,22 @@ After uploading a file, a tile is added to the bottom of the component. This til
...
 - Error
 
-<Preview path="file-uploader-with-file" nude />
+<Preview path="uploaded-files" nude />
```
Comment (nherwegh) [inline src/docs/Components/Form/FileUploader/index.mdx:65]: aren't we supposed to have an error message displayed for the `filename-invalid.jpg`?

File: src/docs/Components/Form/FileUploader/code.mdx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+@import 'components/_c.file-uploader';
+
+@import 'components/_c.fields';
+```
+
+> Note that the import order is important to get the right rendering of the component.
+
+### Usage
+
+When using the fileuploader as part of a form, you must make sure to respect the following structure:
```
Comment (caqueste) [inline src/docs/Components/Form/FileUploader/code.mdx:110]: ~~fileuploader~~ file uploader

File: src/docs/Components/Form/FileUploader/code.mdx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+<span class="mc-field__help">
+  Help text
+</span>
+```
+
+<Preview path="help-text" />
+
+#### Requirement
+
+When the fileuploader is mandatory, you must specify it to the user.
```
Comment (caqueste) [inline src/docs/Components/Form/FileUploader/code.mdx:156]: ~~fileuploader~~ file uploader

File: src/docs/Components/Form/FileUploader/code.mdx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  Label
+  <span class="mc-field__requirement">
+    mandatory
+  </span>
+</label>
+```
+
+<Highlight theme="tips" title="Semantic rule">
+
+  When the fileuploader is mandatory the `required` attribute must be added on the `mc-fileuploader__input` element
```
Comment (caqueste) [inline src/docs/Components/Form/FileUploader/code.mdx:171]: ~~fileuploader~~ file uploader

File: src/docs/Components/Form/FileUploader/code.mdx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+</Highlight>
+
+<Preview path="requirement" />
+
+### Validation
+
+#### Invalid
+
+To indicate that your fileuploader is not valid, several steps are to be implemented:
```
Comment (caqueste) [inline src/docs/Components/Form/FileUploader/code.mdx:181]: ~~fileuploader~~ file uploader

File: src/docs/Components/Form/FileUploader/index.mdx (hunk 3 above)
Comment (caqueste) [reply src/docs/Components/Form/FileUploader/index.mdx:65]: @nherwegh Yes you're right. I will fix the preview ;)

**Pull request merged**